src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,36 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import java.util.Objects;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidRegistrationDataException("User can't be null.");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidRegistrationDataException("Sorry, a user with that "
+                    + "login already exists.");
+        }
+
+        if (Objects.isNull(user.getLogin()) || user.getLogin().length() < 6) {
+            throw new InvalidRegistrationDataException("Login must have at least 6 characters.");
+        }
```

Comment:
Consider splitting the condition into separate checks for null and length to improve readability and provide more specific error messages. This aligns with the checklist recommendation to avoid creating one big if with all conditions.